packages/jest-haste-map/src/index.ts

```diff
@@ -649,17 +679,16 @@ class HasteMap extends EventEmitter {
       }
     }
 
-    return Promise.all(promises)
-      .then(() => {
-        this._cleanup();
-        hasteMap.map = map;
-        hasteMap.mocks = mocks;
-        return hasteMap;
-      })
-      .catch(error => {
-        this._cleanup();
-        return Promise.reject(error);
-      });
+    try {
+      await Promise.all(promises);
+      this._cleanup();
+      hasteMap.map = map;
+      hasteMap.mocks = mocks;
+      return hasteMap;
+    } catch (error) {
```

Comment:
could do


```ts
} finally {
  this._cleanup();
}
```

instead of the `catch` (and remove it from the happy path as well). Not sure if it's better or not?

Reply:
Since the `catch` actually throws the error, it wouldn't make it to the `finally` block in this case. But yeah, the code duplication (even just calling the method) annoys me a little too. I don't see a way around it, though.